sky/spot/recovery_strategy.py

```diff
@@ -211,37 +196,47 @@ def recover(self) -> float:
             logger.info('Ignoring the job cancellation failure; the spot '
                         'cluster is likely completely stopped. Recovering.')
 
-        # Add region constraint to the task, to retry on the same region
-        # first.
-        task = self.dag.tasks[0]
-        resources = list(task.resources)[0]
-        original_resources = resources
-
-        launched_cloud = handle.launched_resources.cloud
-        launched_region = handle.launched_resources.region
-        new_resources = resources.copy(cloud=launched_cloud,
-                                       region=launched_region)
-        task.set_resources({new_resources})
-        # Not using self.launch to avoid the retry until up logic.
-        launched_time = self._launch(raise_on_failure=False)
-        # Restore the original dag, i.e. reset the region constraint.
-        task.set_resources({original_resources})
-        if launched_time is not None:
-            return launched_time
+        while True:
+            # Add region constraint to the task, to retry on the same region
+            # first.
+            task = self.dag.tasks[0]
+            resources = list(task.resources)[0]
+            original_resources = resources
+
+            launched_cloud = handle.launched_resources.cloud
+            launched_region = handle.launched_resources.region
+            new_resources = resources.copy(cloud=launched_cloud,
+                                        region=launched_region)
+            task.set_resources({new_resources})
+            # Not using self.launch to avoid the retry until up logic.
```

Comment:
Can remove this and L225.

Reply:
Since we are still using the `self._launch` here to avoid the retry until up logic in `self.launch`, I feel like it is better to keep the comment as a reminder.

Reply:
Makes sense!